docs/releases/upgrading/upgrade-v12.0.md

```diff
@@ -0,0 +1,164 @@
+
+!!! warning "Risk of losing data"
+
+    Your virtual env folder contains uploaded files in InvenioRDM, in `var/instance/data`.
+    If you need to keep such files, make sure you copy them over to the new virtual env in the same location.
+
+### Upgrade InvenioRDM
+
+(TODO: should we write that 3.9<=X<=3.12 is possible?)
+Make sure that your virtual env is now running with Python 3.9.
+
+Upgrade the RDM version:
+
+```bash
+cd <my-site>
+# Upgrade to InvenioRDM v12
+invenio-cli packages update 12.0.0
+pipenv uninstall flask-babelex
+# Re-build assets
+invenio-cli assets build
+```
+
+Optionally, update the file `<my-site>/Pipfile`. Attention: this action might lead to
+installing unwanted pre-releases of other packages.
+
+```diff
+[packages]
+---invenio-app-rdm = {extras = [...], version = "~=11.0.0"}
++++invenio-app-rdm = {extras = [...], version = "~=12.0.0"}
+
+[pipenv]
+allow_prereleases = true
+```
+
+### Database migration
+
+Execute the database migration:
+
+```bash
+# Execute the database migration
+invenio alembic upgrade
+```
+
+### Declare usage statistics processing queues
+
+```shell
+invenio queues declare
+```
+
+### Update indices mappings
+
+```shell
+invenio index update communities-communities-v1.0.0
+invenio index update rdmrecords-drafts-draft-v5.0.0
+invenio index update rdmrecords-records-record-v5.0.0
+```
+
+CAVEAT: this is not working because of an permission problem in
+[invenio-search](https://github.com/inveniosoftware/invenio-search/blob/d8b23ecf48f63d8d313f90fd4618a480e15fbd7b/invenio_search/ext.py#L448).
+The only way to solve it is to destroy the index and reinit and
+rebuild it from scratch. Yes the statistics will be only in the stored only in
+the index but the statistic indices are created by templates and therefore not
+affected by `invenio index destroy --yes-i-know`. The statistic indices will
+remain in the list of indices and are totally functional also after a reinit.
+
+
+### Data migration
+
+Execute the data migration, note that there is no need to re-index the data:
+
+```bash
+pipenv run invenio shell $(find $(pipenv --venv)/lib/*/site-packages/invenio_app_rdm -name migrate_11_0_to_12_0.py)
+```
+
+### New roles
+
+```bash
+invenio roles create administration-moderation
+invenio roles create administration
+
+invenio access allow administration-moderation role administration-moderation
+invenio access allow administration-access role administration
+invenio access allow superuser-access role administration
+```
+
+### New configuration variables
+
+```bash
+COMMUNITIES_IDENTITIES_CACHE_REDIS_URL = "URI_TO_REDIS"
+USERS_RESOURCES_ADMINISTRATION_ENABLED = True
+THEME_SITENAME = "Project name for header and UI"
+```
+
+## Big Changes
+
+- remove: dependency of flask-babelex
+- add: concept doi (aka parent doi)
+- add: statistics. (NOTE: statistic is stored in opensearch/libresearch)
+- add: administration panel
+- add: set quota
+- add: branded communities
+
```

Comment:
licenses: The delimiter for the licenses vocabulary file has been changed from ";" to ",". If you use a custom vocabulary file you will need to update the delimiter if you want to update licenses in the future. The default licenses are available [here](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/fixtures/data/vocabularies/licenses.csv). You can now add and update licenses with `pipenv run invenio rdm-records add-to-fixture licenses`